CDCgov/prime-reportstream
# Rick/2356 wa ordering facility

This PR implements WA's request to have SimpleReport's organization_name in their HL7 ORC-21 ordering facility name. 

Test Steps:
1. Download a simple report file, filter for WA results 
2. Route to WA either in a local instance or using the CLI
3. Look at HL7 produced
=> Verify ORC-21 is the school district name or the name

## Changes
- A new Hl7Configuration `useOrderingFacilityName` that allows overriding of the standard value that is set in ORC 21
- A new enum `OrderingFacilityName` with standard, NCES, and organization_name values. 
- Hl7Serializer updated to support the new configuration
- Updates to OpenAPI yml and unit tests 

## Checklist

### Testing

### Process

## Fixes
- #2356

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

No to all

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
